compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3729,6 +3742,13 @@ private BType createArrayAndTupleIntersection(BArrayType arrayType, BTupleType t
             tupleMemberTypes.add(intersectionType);
         }
 
```

Comment:
Shall we extract `arrayType.eType` out to a variable outside the for loop above?